app/controllers/policies_controller.rb

```diff
@@ -57,7 +57,7 @@ def post
   # Returns newly created roles
   def perform(policy_action)
     policy_action.call
-    new_actor_roles = actor_roles(policy_action.new_roles)
+    new_actor_roles = actor_roles(policy_action.loader)
```

Comment:
Ah, no, you want to add `new_roles` to the 3 objects you created and _delegate_ the implementation to the underlying loader.  What we're doing above violates demeter's law, and forces too much knowledge out into the client code.  

You can accomplish the delegation with the Forwardable gem, but here since it's just one method I'd prefer to write it manually.